src/snowcli/cli/nativeapp/manager.py

```diff
@@ -270,8 +274,22 @@ def _apply_package_scripts(self) -> None:
                 raise InvalidPackageScriptError(relpath, e)
 
         # once we're sure all the templates expanded correctly, execute all of them
-        if self.warehouse:
-            self._execute_query(f"use warehouse {self.warehouse}")
+        try:
+            if self.package_warehouse:
+                self._execute_query(f"use warehouse {self.package_warehouse}")
+            elif self._conn.warehouse:
+                self._execute_query(f"use warehouse {self._conn.warehouse}")
```

Comment:
based on our offline discussion, it might be cleaner to deal with only one provider side warehouse, and do this if/else logic above.